test/e2e/framework/.import-restrictions

```diff
@@ -1,269 +1,220 @@
-      - k8s.io/kubernetes/pkg/scheduler/testing
-      - k8s.io/kubernetes/pkg/scheduler/metrics
-      - k8s.io/kubernetes/pkg/scheduler/nodeinfo
       - k8s.io/kubernetes/pkg/scheduler/util
-      - k8s.io/kubernetes/pkg/scheduler/volumebinder
-      - k8s.io/kubernetes/pkg/scheduler
-      - k8s.io/kubernetes/pkg/scheduler/profile
-      - k8s.io/kubernetes/pkg/scheduler/testing
-      - k8s.io/kubernetes/pkg/security/apparmor
-      - k8s.io/kubernetes/pkg/securitycontext
-      - k8s.io/kubernetes/pkg/serviceaccount
-      - k8s.io/kubernetes/pkg/util/async
-      - k8s.io/kubernetes/pkg/util/bandwidth
-      - k8s.io/kubernetes/pkg/util/config
-      - k8s.io/kubernetes/pkg/util/configz
-      - k8s.io/kubernetes/pkg/util/conntrack
-      - k8s.io/kubernetes/pkg/util/env
-      - k8s.io/kubernetes/pkg/util/filesystem
-      - k8s.io/kubernetes/pkg/util/flag
-      - k8s.io/kubernetes/pkg/util/flock
-      - k8s.io/kubernetes/pkg/util/goroutinemap
-      - k8s.io/kubernetes/pkg/util/goroutinemap/exponentialbackoff
-      - k8s.io/kubernetes/pkg/util/hash
-      - k8s.io/kubernetes/pkg/util/ipset
-      - k8s.io/kubernetes/pkg/util/iptables
-      - k8s.io/kubernetes/pkg/util/ipvs
-      - k8s.io/kubernetes/pkg/util/labels
-      - k8s.io/kubernetes/pkg/util/node
-      - k8s.io/kubernetes/pkg/util/oom
-      - k8s.io/kubernetes/pkg/util/parsers
-      - k8s.io/kubernetes/pkg/util/pod
-      - k8s.io/kubernetes/pkg/util/procfs
-      - k8s.io/kubernetes/pkg/util/removeall
-      - k8s.io/kubernetes/pkg/util/resizefs
-      - k8s.io/kubernetes/pkg/util/rlimit
-      - k8s.io/kubernetes/pkg/util/selinux
+      - k8s.io/kubernetes/pkg/scheduler/apis/config
       - k8s.io/kubernetes/pkg/util/slice
-      - k8s.io/kubernetes/pkg/util/sysctl
-      - k8s.io/kubernetes/pkg/util/system
-      - k8s.io/kubernetes/pkg/util/tail
-      - k8s.io/kubernetes/pkg/util/taints
-      - k8s.io/kubernetes/pkg/volume
+      - k8s.io/kubernetes/pkg/util/labels
       - k8s.io/kubernetes/pkg/volume/util
-      - k8s.io/kubernetes/pkg/volume/util/fs
-      - k8s.io/kubernetes/pkg/volume/util/fsquota
-      - k8s.io/kubernetes/pkg/volume/util/recyclerclient
-      - k8s.io/kubernetes/pkg/volume/util/subpath
-      - k8s.io/kubernetes/pkg/volume/util/types
-      - k8s.io/kubernetes/pkg/volume/util/volumepathhandler
-      # TODO: I have no idea why import-boss --include-test-files is yelling about these for k8s.io/kubernetes/test/e2e/framework/providers/kubemark
-      - k8s.io/kubernetes/pkg/apis/authentication
-      - k8s.io/kubernetes/pkg/apis/authentication/v1
-      - k8s.io/kubernetes/pkg/apis/certificates/v1beta1
-      - k8s.io/kubernetes/pkg/apis/storage/v1
-      - k8s.io/kubernetes/pkg/scheduler/internal/cache
+  # prevent any k8s.io/kubernetes/test/ imports outside of this package
+  # Packages are used in the test/
   - selectorRegexp: k8s[.]io/kubernetes/test/
     allowedPrefixes:
+      - k8s.io/kubernetes/test/conformance/testdata
```

Comment:
I don't think we need to (and shouldn't) list every single package here.

What we want to say is "anything under test/e2e/framework may import anything under test". Or at least I can't think of something that shouldn't be allowed.

If that is the goal, then can't we just use this:
```
  # Everything under k8s.io/kubernetes/test/ may be used.
  - selectorRegexp: k8s[.]io/kubernetes/test/
    allowedPrefixes:
    - ""
```

Or even leave the rule out entirely?

Reply:
According to the first comment, I have listed down all the packages which are being used in the test package. Yes, the list is too long so it is better to only keep prefix which are being used.

Second, in the */test package we can use the suggested changes as most of the import are used.

I will make the changes.